modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs

```diff
@@ -69,21 +106,168 @@ Compilation compilation
             SymbolDisplayFormat.FullyQualifiedFormat
                 .WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted);
 
-        public static string FullQualifiedName(this INamedTypeSymbol symbol)
+        public static string FullQualifiedName(this ITypeSymbol symbol)
             => symbol.ToDisplayString(NullableFlowState.NotNull, FullyQualifiedFormatOmitGlobal);
 
+        public static string NameWithTypeParameters(this INamedTypeSymbol symbol)
+        {
+            return symbol.IsGenericType ?
+                string.Concat(symbol.Name, "<", string.Join(", ", symbol.TypeParameters), ">") :
+                symbol.Name;
+        }
+
         public static string FullQualifiedName(this INamespaceSymbol namespaceSymbol)
             => namespaceSymbol.ToDisplayString(FullyQualifiedFormatOmitGlobal);
+
+        public static string SanitizeQualifiedNameForUniqueHint(this string qualifiedName)
+            => qualifiedName
+                // AddSource() doesn't support angle brackets
+                .Replace("<", "(Of ")
+                .Replace(">", ")");
+
+        public static bool IsGodotExportAttribute(this INamedTypeSymbol symbol)
+            => symbol.ToString() == GodotClasses.ExportAttr;
+
+        public static bool IsGodotSignalAttribute(this INamedTypeSymbol symbol)
+            => symbol.ToString() == GodotClasses.SignalAttr;
+
+        public static bool IsGodotClassNameAttribute(this INamedTypeSymbol symbol)
+            => symbol.ToString() == GodotClasses.GodotClassNameAttr;
+
+        public static bool IsSystemFlagsAttribute(this INamedTypeSymbol symbol)
+            => symbol.ToString() == GodotClasses.SystemFlagsAttr;
+
+        public static GodotMethodData? HasGodotCompatibleSignature(
+            this IMethodSymbol method,
+            MarshalUtils.TypeCache typeCache
+        )
+        {
+            var retSymbol = method.ReturnType;
+            var retType = method.ReturnsVoid ?
+                null :
+                MarshalUtils.ConvertManagedTypeToMarshalType(method.ReturnType, typeCache);
+
+            if (retType == null && !method.ReturnsVoid)
+                return null;
+
+            var parameters = method.Parameters;
+
+            var paramTypes = parameters
+                // Currently we don't support `ref`, `out`, `in`, `ref readonly` parameters (and we never may)
+                .Where(p => p.RefKind == RefKind.None)
+                // Attempt to determine the variant type
+                .Select(p => MarshalUtils.ConvertManagedTypeToMarshalType(p.Type, typeCache))
+                // Discard parameter types that couldn't be determined (null entries)
+                .Where(t => t != null).Cast<MarshalType>().ToImmutableArray();
+
+            // If any parameter type was incompatible, it was discarded so the length won't match
+            if (parameters.Length > paramTypes.Length)
+                return null; // Ignore incompatible method
+
```

Comment:
Here's a potential fix for [what I think is a bug](https://github.com/godotengine/godot/pull/64089#issuecomment-1214207557)

```suggestion
            // Generic methods are also incompatible
            if (method.IsGenericMethod)
                return null;
```